src/Sentry/SentryHttpMessageHandler.cs

```diff
@@ -85,7 +86,12 @@ protected internal override void HandleResponse(HttpResponseMessage response, IS
         _failedRequestHandler?.HandleResponse(response);
 
         // This will handle unsuccessful status codes as well
-        var status = SpanStatusConverter.FromHttpStatusCode(response.StatusCode);
-        span?.Finish(status);
+        if (span is not null)
+        {
+            span.SetExtra(OtelSemanticConventions.AttributeHttpRequestMethod, method);
+            span.SetExtra(OtelSemanticConventions.AttributeHttpResponseStatusCode, (int)response.StatusCode);
```

Comment:
Note that I thought about putting this on the scope. In a scenario where the HttpClient is inside an ASP.NET Core Endpoint though, the Request.Method and Response.StatusCode already refer to the method and status code of the HttpRequest to the webserver. 

Another option might be to push a scope when the span gets created in `SentryMessageHandler.SendAsync`, set the Request and Response info on the Scope as context there, and then pop the scope once the span finishes. 

Maybe that's a cleaner solution? It would mean we could consolidate the logic to set these attributes for outbound requests in the `SentryMessageHandler` as well (rather than having it duplicated, as it currently is, across `SentryHttpMessageHandler` and `SentryGraphQLHttpMessageHandler`).

Reply:
Yeah agreed, I'll move it.

Reply:
Ah, actually... this is one reason it can't be done when the span gets created:
```
span.SetExtra(OtelSemanticConventions.AttributeHttpResponseStatusCode, (int)response.StatusCode);
```

We don't have the response code until later:

https://github.com/getsentry/sentry-dotnet/blob/b5ab31abfe4e9d0a891d260edb91faf03fecc922/src/Sentry/SentryMessageHandler.cs#L88-L99

I'll move it into the `SentryMessageHandler` though. If the logic ever diverges, between the different concrete message handlers, we can always push it back out to the concrete classes.

Reply:
Actually, these things are already a bit different... I think better to just leave them separate.

This is what the SentryHttpMessageHandler is doing:
https://github.com/getsentry/sentry-dotnet/blob/b5ab31abfe4e9d0a891d260edb91faf03fecc922/src/Sentry/SentryHttpMessageHandler.cs#L91-L94

And this is what the SentryGraphQLMessageHandler is doing:
https://github.com/getsentry/sentry-dotnet/blob/b5ab31abfe4e9d0a891d260edb91faf03fecc922/src/Sentry/SentryGraphQLHttpMessageHandler.cs#L91-L96

Not enough in common, I don't think, to justify refactoring to a common base method.